platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -14,92 +14,77 @@
 import org.osgi.framework.BundleContext;
 import org.osgi.framework.Constants;
-import org.osgi.framework.FrameworkUtil;
-import org.osgi.framework.InvalidSyntaxException;
 import org.osgi.framework.ServiceReference;
+import org.osgi.service.http.whiteboard.HttpWhiteboardConstants;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * {@link DelegateServletFilter} is meant to detect any ServletFilters
- * and call their {@link FilterChain}. If none are found it becomes a pass
- * through filter.
- *
+ * {@link DelegateServletFilter} is meant to detect any {@link SecurityFilter}s and doFilter through
+ * those before continuing on its {@link FilterChain}. If no {@link SecurityFilter}s are found, it
+ * becomes a pass-through filter.
  */
 public class DelegateServletFilter implements Filter {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(DelegateServletFilter.class);
 
-    private FilterConfig filterConfig;
+    final TreeSet<ServiceReference<SecurityFilter>>
+            sortedAndInitializedSecurityFilterServiceReferences;
 
-    protected BundleContext getContext() {
-        Bundle cxfBundle = FrameworkUtil.getBundle(DelegateServletFilter.class);
-        if (cxfBundle != null) {
-            return cxfBundle.getBundleContext();
-        }
-        return null;
+    // used to only init SecurityFilters after the DelegateServletFilter has been initialized with a filterConfig
+    final Set<ServiceReference<SecurityFilter>> notYetInitializedSecurityFilterServiceReferences;
+
+    private Boolean hasBeenInitialized;
+
+    private final BundleContext bundleContext;
+
+    FilterConfig filterConfig;
+
+    public DelegateServletFilter(final BundleContext bundleContext) {
+        this.bundleContext = bundleContext;
+        sortedAndInitializedSecurityFilterServiceReferences = new TreeSet<>();
+        notYetInitializedSecurityFilterServiceReferences = new HashSet<>();
+        hasBeenInitialized = false;
     }
 
     @Override
     public void destroy() {
+        // do nothing
     }
 
     @Override
     public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
             FilterChain filterChain) throws IOException, ServletException {
-
-        Collection<ServiceReference<Filter>> referenceCollection = null;
-        BundleContext context = getContext();
-
-        if (null != context) {
-            try {
-                referenceCollection = context.getServiceReferences(Filter.class, null);
-            } catch (InvalidSyntaxException ise) {
-                LOGGER.debug("Should not get this exception as there is no filter being passed.");
-            }
-        }
-
-        if (referenceCollection != null && !referenceCollection.isEmpty()) {
-            LOGGER.debug("Found {} filter(s), now filtering...", referenceCollection.size());
-
+        if (!CollectionUtils.isEmpty(sortedAndInitializedSecurityFilterServiceReferences)) {
```

Comment:
Could we just add another check here to see if everything is initialized and do a pass through if not? That way we don't need the notYetInitializedSecurityFilterServiceReferences set and can just put all references directly in the sortedAndInitializedSecurityFilterServiceReferences when adding filters and reduce the complexity of the class.